src/features/User/userSlice.ts

```diff
@@ -1,15 +1,9 @@
 import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
 import axios from 'axios';
-import { UserType } from 'types';
+import { UserStateType, UserType } from 'types';
 const user = localStorage.getItem('user');
```

Comment:
suggestion - restoring logged In state can be extracted to its own thunk and called in root component.